src/generic/Matrix.jl

```diff
@@ -5015,13 +5015,11 @@ function (a::MatSpace{T})(b::Mat{T}) where {T <: RingElement}
    return b
 end
 
-function (a::MatSpace{T})(b::AbstractArray{T, 2}) where T <: RingElement
+function (a::MatSpace{T})(b::Array{T, 2}) where T <: RingElement
    R = base_ring(a)
    _check_dim(a.nrows, a.ncols, b)
-   for i = 1:a.nrows
-      for j = 1:a.ncols
-         b[i, j] = R(b[i, j])
-      end
+   if (a.nrows != 0 || a.ncols != 0)
```

Comment:
it should be `if (a.nrows != 0 && a.ncols != 0)`, but even better `if !isempty(b)`

Reply:
Thanks. Fixed.